node/src/reactor/main_reactor/upgrade_shutdown.rs

```diff
@@ -79,7 +79,11 @@ impl MainReactor {
     pub(super) fn upgrade_shutdown_instruction(
         &self,
         effect_builder: EffectBuilder<MainEvent>,
+        time_switched: Timestamp,
     ) -> UpgradeShutdownInstruction {
+        if time_switched.elapsed() > self.shutdown_for_upgrade_timeout {
+            return self.schedule_shutdown_for_upgrade(effect_builder);
```

Comment:
So, after the timeout we schedule shutdown for upgrade, which - in turn - will still wait additional `DELAY_BEFORE_SHUTDOWN`.

It's not harmful, but just wanted to point it out.